binlog-stream/test/io/laserdisc/mysql/binlog/compaction/CompactTest.scala

```diff
@@ -0,0 +1,148 @@
+package io.laserdisc.mysql.binlog.compaction
+
+import io.circe.parser.parse
+import io.laserdisc.mysql.binlog.event.EventMessage
+import org.scalatest.OptionValues
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+
+class CompactTest extends AnyWordSpec with OptionValues with Matchers {
```

Comment:
This tests the the compaction of a provided Seq of events, but not that the compaction occurs at a transactional boundary...  Is there a way to tie this in somehow with the TransactionStateTest?

Reply:
Well, i decided to make compaction on arbitrary sequence of events, because probably one may want to do it outside transaction boundary.
However current public API provides transaction compaction stream, which uses transaction package and bounds compactions to transaction.
So current compaction api is very generic, and any client can use it on any desired boundary